### PR TITLE
Issue #40 Simplified PaymentInfoConverters and added missing Types

### PR DESCRIPTION
**Pull request by BLangendorf**

### CODE REVIEW

File: src/main/java/org/killbill/billing/plugin/adyen/client/payment/converter/impl/NullObjectConverter.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 public class NullObjectConverter implements PaymentInfoConverter {
 
     @Override
-    public PaymentType getPaymentType() {
+    public boolean supportsPaymentType(PaymentType type) {
         throw new UnsupportedOperationException("method is not supported!");
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/client/payment/converter/impl/NullObjectConverter.java]: Maybe add the `type` in the error message?

**Comment by pierre** [comment]
:+1:

File: src/main/java/org/killbill/billing/plugin/adyen/client/payment/converter/impl/NullObjectConverter.java (hunk 1 above)
Comment (BLangendorf) [inline src/main/java/org/killbill/billing/plugin/adyen/client/payment/converter/impl/NullObjectConverter.java]: Well, the `UnsupportedOperationException` means this operation is not supported at all, not just for some values of `type` … but seeing the `type` might help debugging --> I'll add it :smiley:

Comment (BLangendorf) [inline src/main/java/org/killbill/billing/plugin/adyen/client/payment/converter/impl/NullObjectConverter.java]: Commit 1e749658af9300dbfb9bc175cc56e89b033120dc

Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/client/payment/converter/impl/NullObjectConverter.java]: Yup, it's mainly for debugging if we end-up using that converter for some reason (see [here](https://github.com/killbill/killbill-adyen-plugin/blob/d45e3c060373ac12e53c76b89d919573ea7d7d7f/src/main/java/org/killbill/billing/plugin/adyen/client/payment/builder/PaymentRequestBuilder.java#L68)).

**Pull request merged**